### PR TITLE
Alright, lots of things. 

### PR DESCRIPTION
**Pull request by Portentum-V**
1) Moved 'dry_run_enabled' into the sd_api_call; 
2) merged mp3, and opus handling into a single function;
3) added a function to check if Author == AI; 
4) Created a 'main' function to handle the program flow; 
5) Added some example argparse, it should load defaults from the given json config file and allow the use of command line switches (DRAFT)

### CODE REVIEW

File: smeargle.py [hunk 1]
```diff
@@ -48,6 +52,9 @@ def sd_api_call():
     # Increment the total api call counter
     api_call_count += 1
 
+    if dry_run_enabled:
```
Comment (Portentum-V) [inline smeargle.py:55]: This way you don't have to increment the api call outside the function if its a dry run

File: smeargle.py [hunk 2]
```diff
@@ -177,103 +184,109 @@ def sd_api_call():
         image.save(f'test_image_output/{api_call_count}.png', pnginfo=metadata)
         print(f"image successfully created, tagged, and moved to test_image_output/{api_call_count}.png. sleeping for {sleep_timer}s")
     else:
-        image.save(f'{dirpath}\cover.png', pnginfo=metadata)
-        print(f"image successfully created, tagged, and moved to {dirpath}\cover.png. sleeping for {sleep_timer}s\n")
+        image.save(f'{dirpath}\\cover.png', pnginfo=metadata)
+        print(f"image successfully created, tagged, and moved to {dirpath}\\cover.png. sleeping for {sleep_timer}s\n")
```
Comment (Portentum-V) [inline smeargle.py:188]: Pretty sure this needs an extra slash, was it working before?

File: smeargle.py [hunk 3]
```diff
@@ -177,103 +184,109 @@ def sd_api_call():
...
+    im.load()
+    return im.info['Author'] == name
+
+def check_and_generate(dirpath, musicfile, music_extension):
+    musicfilepath = os.path.join(dirpath, musicfile)
+    print(f"{music_extension} found: {musicfilepath}")
+    
+    tag = music_tag.load_file(f"{musicfilepath}")
+
+    # check if there is any artwork attached to the .mp3 file ("artwork" tag)
```
Comment (Portentum-V) [inline smeargle.py:205]: As far as I could tell, the only difference between .mp3 and opus was this extra check for tag['artwork']

File: smeargle.py (hunk 2 above)
Comment (ShiningLava) [reply smeargle.py:188]: Yes it was working before. It threw errors about improper escapes but the script did run despite the errors. This was for Windows and I just copy pasted the path from File Explorer, hence the single slash. Good fix, will include.

File: smeargle.py (hunk 3 above)
Comment (ShiningLava) [reply smeargle.py:205]: Correct. The extra check for artwork tags is done differently between the two file types. mp3 is very easy to detect artwork as it is an easily readable tag in the file that has had wide support for a long time. But opus is different, and more complicated (see https://wiki.xiph.org/VorbisComment#METADATA_BLOCK_PICTURE). The original goal for .opus was to encode the stable diffusion image into the proper format for adding to the METADATA_BLOCK_PICTURE tag, but I wasn't able to get it working well enough to publish. 

I decided to skip the typical opus artwork check in favor or just checking if smeargle AI artwork already exists in the form of cover.png. 

Good fix, will include.

File: smeargle.py [hunk 4]
```diff
@@ -177,103 +184,109 @@ def sd_api_call():
...
+    parser.add_argument("--icon_path", type=str, help="Path to icon. If you use a custom icon, it should be size 150x150. Currently, only .png files supported")
+    parser.add_argument("--sleep_timer", type=int, help="The number of seconds the script will sleep after each image created. Helps with GPU temperatures, but slows the process down")
+    parser.add_argument("--dry_run_enabled", type=bool, help="Skip generating new images, instead will just analyze the music_directory and print results in the console")
+    parser.add_argument("--regenerate_ai_artwork", type=bool, help="Every cover.png in the music_directory or subdirectories that has the 'Author' tag set to 'AI' will be deleted and regenerated")
+    parser.add_argument("--blur_level", type=int, help="0 = No blur effect applied, 5 = high blur. Blur can make SD images appear less ugly at a glance")
+    parser.add_argument("--stable_diffusion_address", type=str, help="Use API calls to public or paid instances at your own risk")
+    
+    parser.add_argument("--prompt", type=str)
+    parser.add_argument("--test_image_output_folder", type=str)
```
Comment (ShiningLava) [inline smeargle.py:269]: I believe test_image_output_folder should expect a boolean. In future revisions smeargle could allow for custom directory locations, but currently does not

Comment (ShiningLava) [reply smeargle.py:269]: Nevermind I should have just read the next line lol

**Pull request merged**